src/xmipp/libraries/reconstruction/flexible_alignment.cpp

```diff
@@ -434,17 +447,18 @@ int partialpfunction(Matrix1D<double> &Parameters,
 		// Extract atom type and position
 		// Typical line:
 		// ATOM    909  CA  ALA A 161      58.775  31.984 111.803  1.00 34.78
-		std::string atom_type = line.substr(13, 2);
-		double x = textToFloat(line.substr(30, 8));
-		double y = textToFloat(line.substr(38, 8));
-		double z = textToFloat(line.substr(46, 8));
+		const auto& atom = pdb.atomList[nAtom];
+		char atom_type = atom.atomType;
+		double xPos = atom.x;
+		double yPos = atom.y;
+		double zPos = atom.z;
 
 		// Correct position
-		coord_gaussian[0] = (x - centerOfMass(0))
+		coord_gaussian[0] = (xPos - centerOfMass(0))
```

Comment:
xPos, yPos, and zPos because x, y, and z produced code smells:
Shadowing local variable